cli/cli.go

```diff
@@ -19,24 +20,28 @@ import (
 	"github.com/notaryproject/notation-plugin-framework-go/plugin"
 )
 
+// CLI struct is used to create an executable for plugin.
 type CLI struct {
 	name   string
 	pl     plugin.Plugin
 	logger log.Logger
 }
 
 // New creates a new CLI using given plugin
-func New(executableName string, pl plugin.Plugin) *CLI {
-	return NewWithLogger(executableName, pl, &discardLogger{})
+func New(pl plugin.Plugin) (*CLI, error) {
+	return NewWithLogger(pl, &discardLogger{})
 }
 
 // NewWithLogger creates a new CLI using given plugin and logger
-func NewWithLogger(executableName string, pl plugin.Plugin, l log.Logger) *CLI {
+func NewWithLogger(pl plugin.Plugin, l log.Logger) (*CLI, error) {
+	if pl == nil {
+		return nil, errors.New("plugin cannot be nil")
+	}
+
 	return &CLI{
-		name:   executableName,
 		pl:     pl,
 		logger: l,
-	}
+	}, nil
```

Comment:
We have the `name` field defined in `CLI` struct. In `NewWithLogger`, we should get the plugin's metadata and fill `CLI.name` with `md.Name`.

Reply:
Removed the field as its unused.